### PR TITLE
feat(challenges): Adds do...while challenge to basic JS challenges

### LINKED ISSUE #14948: Beta Suggestion: do - while loop
Challenge [iterate-with-javascript-while-loops](http://beta.freecodecamp.com/en/challenges/basic-javascript/iterate-with-javascript-while-loops) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

Since the loop mechanism is taught differently, this seems to be a great place to teach the do while loop.

### PR DESCRIPTION
**Pull request by quatridunull**
Closes #14948

#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #14948

#### Description
This adds a new challenge to the basic javascript challenges. It introduces the do...while loop syntax.

### CODE REVIEW

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json [hunk 1]
```diff
@@ -5393,6 +5393,54 @@
         }
       }
     },
+    {
+      "id": "xf2222c1x11fedafarb1blll",
```
Comment (raisedadead) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: This id looks a bit off, do we not have a standard way of generating them for new challenges? @QuincyLarson @Hkuz if you guys can confirm?

**Review by raisedadead — COMMENTED** [review]
This id looks a bit off, do we not have a standard way of generating them for new challenges? @QuincyLarson @HKuz if you guys can confirm?

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json (hunk 1 above)
Comment (QuincyLarson) [reply seed/challenges/02-javascript-algorithms-and-data-structures/basic-javascript.json]: @raisedadead Yes - use this code instead. I generated it using the `ObjectId()` command in the MongoDB shell:

`5a2efd662fb457916e1fe604`

**Comment by raisedadead** [comment]
@QuincyLarson I have updated the id, the challenge it self is not changed. Can you give this a merge?

**Comment by quatridunull** [comment]
Hi. Sorry I haven't looked at the comments until now. I thought the PR had been rejected after a month and forgot about it. Do you still need me to amend the commit and update the PR? Are there any other changes needed?

**Comment by raisedadead** [comment]
Hi @quatridunull 

Thanks for getting back and taking time to contribute to the project. 
Yup, we are sorry about the delay in the QA.

I have fixed up the necessary id of the challenge :smile: !

We just need to merge, someone from the team will do it, since I have a commit on the PR as per standard practice.

Thanks once again. And we hope you keep contributing in future! 

Cheers and happy contibuting!

**Comment by QuincyLarson** [comment]
@quatridunull OK - I've merged this. Thanks again for creating this challenge, and congratulations on your first merged pull request to the freeCodeCamp repo! We would welcome many more code contributions from you :)

**Comment by quatridunull** [comment]
Ok, cool. Thanks!

**Pull request merged**